File: python/roman-numerals/roman_numerals.py
def numeral(number):
    ones = ["", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"]
    tens = ["", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"]
    hundreds = ["", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"]
    result = []
    while number >= 1000:
        result.append('M')
        number -= 1000
    result.append(hundreds[number // 100])
    number = number % 100
    result.append(tens[number // 10])
    number = number % 10
    result.append(ones[number])
    return ''.join(result)

File: go/roman-numerals/roman_numerals.go
package romannumerals

import (
	"errors"
	"strings"
)

const testVersion = 3

var huns = []string{"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"}
var tens = []string{"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"}
var ones = []string{"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"}

func ToRomanNumeral(n int) (string, error) {
	if n <= 0 || n >= 4000 {
		return "", errors.New("Out of range")
	}
	results := []string{}
	for n >= 1000 {
		results = append(results, "M")
		n -= 1000
	}
	results = append(results, huns[n/100])
	n = n % 100

	results = append(results, tens[n/10])
	n = n % 10

	results = append(results, ones[n])
	return strings.Join(results, ""), nil
}
